strings/base_coroutine_foundation.h

```diff
@@ -801,4 +801,43 @@ WINRT_EXPORT namespace winrt
         impl::check_status_canceled(shared->status);
         co_return shared->result.GetResults();
     }
+
+    template <typename Delegate>
+    struct async_event : impl::event_base<Delegate>
```

Comment:
The name `async_event` may be too ambiguous. I can think of six possible meanings!

1. This is a an object that you `co_await` and wait for it to be signaled. Sort of like an "awaitable event" (where "event" is used in the kernel sense).
2. This is a WinRT event which invokes its callbacks sequentially on a background thread. Used if you want to protect the main thread from hanging on a stuck handler. You can await the initiation of the handlers (but cannot await their completions).
3. This is a WinRT event which invokes its callbacks in parallel on background threads. Used if you want to protect one stuck handler from interfering with all the other handlers.  You can await the initiation of the handlers (but cannot await their completions).
4. This is a WinRT event which invokes its callbacks sequentially on the main thread, then awaits them all, and you can await the aggregate.
5. This is a WinRT event which invokes its callbacks sequentially on a background thread, then awaits them all, and you can await the aggregate.
6. This is a WinRT event which invokes its callbacks in parallel on a background thread, then awaits them all, and you can await the aggregate. Calling in parallel means that one stuck handler does not prevent the other handlers from receiving the callback.

It looks like this is option 3.

Reply:
@jonwis asked for an async version of `winrt::event` - this is what I came up with. It allows handlers to execute in parallel to avoid any single handler blocking progress on the rest. It also allows the event source to continue without waiting for all the handlers to execute, although it does allow the event source to wait for their completion.

Reply:
See https://github.com/microsoft/cppwinrt/issues/724#issuecomment-783562037 for an example.